stripe/api_requestor.py

```diff
@@ -286,16 +311,20 @@ def request_headers(self, api_key, method):
 
     def request_raw(
         self,
-        method,
-        url,
-        params=None,
-        supplied_headers=None,
-        is_streaming=False,
-    ):
+        method: str,
+        url: str,
+        params: Optional[Mapping[str, Any]] = None,
+        supplied_headers: Optional[Mapping[str, str]] = None,
+        is_streaming: bool = False,
+    ) -> Tuple[object, int, Mapping[str, str], str]:
         """
         Mechanism for issuing an API call
         """
 
+        supplied_headers_: Optional[Dict[str, str]] = (
```

Comment:
Should we type the parameter as dict?

Reply:
`Mapping[str, str]` is really a better type. We want to guarantee that the implementation won't mutate the headers that the caller passes in. And also the caller should be able to pass immutable mappings too.

We have to create this temporary dict because (surprise!) we actually *were* mutating. So this does constitute a (almost certainly welcome) behavior change. I should test this out and call it out in the PR description, though.

Reply:
Pulled it out to a separate PR https://github.com/stripe/stripe-python/pull/1091

But it looks like this might actually be load-bearing mutation... I should probably refresh my understanding of how multipart content-type works.